src/app/modals/search-modal/search-modal.component.ts

```diff
@@ -266,6 +266,12 @@ export class SearchModal implements OnInit, AfterViewInit {
     if(filterParams["endDate"]) {
       queryParams["endDate"] = filterParams["endDate"]
     }
+    if (filterParams['geography']) {
```

Comment:
Could be extracted to a smaller method and be unit tested, this function is already bloated.